src/tests/core.py

```diff
@@ -274,6 +276,52 @@ def test_old_knownnodes_pickle(self):
         finally:
             cleanup(files=('knownnodes.dat',))
 
+    @staticmethod
+    def generate_random_address():
+        """Generating random address"""
+        import queues
+        streamNumberForAddress = 1
+        queues.addressGeneratorQueue.put((
+            'createRandomAddress', 4, streamNumberForAddress,
+            "test1", 1, "", False))
```

Comment:
you should also wait for the address to be generated Look at https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/bitmessageqt/support.py#L87 to see how to do it.